sk_sidekick/lib/src/commands/lock_dependencies_command.dart

```diff
@@ -43,7 +42,11 @@ class LockDependenciesCommand extends Command {
       }
     }
 
-    final pubGet = systemDart(['pub', 'get'], workingDirectory: package.root);
+    final pubGet = systemDart(
+      ['pub', 'get'],
+      workingDirectory: package.root,
+      nothrow: true,
+    );
     if (pubGet != 0) {
       throw "Couldn't get dependencies in ${package.root}";
     }
```

Comment:
We use `nothrow: true` a lot to add better error messages. We should consider a better API for this.

like: 

```dart
systemDart(['pub', 'get'], errorMessage: "Couldn't get dependencies in ${package.root}");
systemDart(['pub', 'get'], errorMessage: (code) => "Couldn't get dependencies in ${package.root} (exit code $code)");
```